feat: add password reset process

Maaayhan opened the pull request:
### Task description:

Added the process of resetting user password:
 - implemented mutation 'forgotPassword' and 'resetPassword'
 - refactored the email service for reuse
- added corresponding validation pipes to two mutations to verify the validity of data input

### Work flow:

- The user enters their email address on the "Forgot Password" page.
- A reset password email should be sent to the user's provided email address (the email contains a link, valid only 1 hours)
- The user clicks the link to navigate to the password reset page and enters a new password.
- The user's password should be update after the link is successfully verified.
- Any errors encountered during any of these steps will return the error code and message.

### Completed:

**Call forgotPassword with 'email' to update reset password token :**
<img width="1393" alt="image" src="https://github.com/user-attachments/assets/20586d37-2bf5-41cc-83dd-867e82a89af2">

**Call resetPassword with the token and the new password:**
<img width="1428" alt="image" src="https://github.com/user-attachments/assets/27255c27-c9f4-4623-8d78-c29765404742">
<img width="1220" alt="image" src="https://github.com/user-attachments/assets/1a859c01-0390-43fd-a831-6b0d34400c96">

**Reset password token should be deleted after the process:**
<img width="1298" alt="image" src="https://github.com/user-attachments/assets/2d1463fa-5e6e-405c-a4b2-d7b09d6b58cc">

**Verify the password is updated successfully using Login mutation:**
old password:
<img width="1220" alt="image" src="https://github.com/user-attachments/assets/bb4f1a9e-a569-43a2-bd71-49f12710564d">

new password:
<img width="1397" alt="image" src="https://github.com/user-attachments/assets/98c84678-bc1e-4c04-ab7b-38e4028b68c4">

### Unit test:
<img width="479" alt="image" src="https://github.com/user-attachments/assets/43dc2bfd-a563-45b7-8528-a58c014fd8cb">

Resolve CP-28

Chris-Wang commented on src/modules/auth/auth.service.ts:
```diff
@@ -212,4 +214,96 @@ export class AuthService {
+
+  async resetPassword(input: ResetPasswordInput): Promise<Result> {
+    try {
+      const { newPassword, resetToken } = input;
```
resetToken should not be inside the input, it's used to be gotten from header.

And validate token should be a step prior to resetPassword process.

Chris-Wang commented on src/common/utils/helpers.ts:
```diff
@@ -82,3 +87,8 @@ export const refSchema = z
   .optional()
   .default(process.env.DEFAULT_REFERENCE || 'Default_Reference_Name');
+
+export const resetTokenSchema = z
```
why a token has a schema?

Maaayhan replied:
yeah it's redundant, we'll use jwt function to check the token's validity. I will remove the schema here.

Chris-Wang commented on src/modules/auth/auth.service.ts:
```diff
@@ -212,4 +214,96 @@ export class AuthService {
+      const res = await this.userService.update(user.id, {
+        resetPasswordToken,
+      });
+      const requestResetUser = await this.userService.findByEmail(email);
```
why there is user query again?

Maaayhan committed:
feat: add password reset process

Added the process of resetting user password:
 - implemented mutation 'forgotPassword' and 'resetPassword'
 - refactored the email service for reuse
- added corresponding validation pipes to two mutations to verify the validity of data input

Resolve CP-28

Chris-Wang reviewed (approved):
LGTM
